src/nnet3/nnet-chain-training.cc

```diff
@@ -68,23 +68,43 @@ void NnetChainTrainer::Train(const NnetChainExample &chain_eg) {
                              &request);
   const NnetComputation *computation = compiler_.Compile(request);
 
-  NnetComputer computer(nnet_config.compute_config, *computation,
+  if (nnet_config.adversarial_training_scale > 0.0) {
```

Comment:
@freewym, you don't seem to have in this PR the stuff where we do the adversarial training with some specified probability for each minibatch.  I think that's important, as it makes it much faster.  Can you add that to the PR before I review it in detail?